solarforecastarbiter/validation/tasks.py

```diff
@@ -37,6 +50,58 @@ def _solpos_night(observation, values):
     return solar_position, night_flag
 
 
+def _resample_date_range(interval_length, closed, freq, values):
+    # consider moving this to utils
+    data_start, data_end = values.index[0], values.index[-1]
+    if closed == 'left':
+        data_end += interval_length
+    elif closed == 'right':
+        data_start -= interval_length
+    else:
+        raise ValueError("closed must be 'left' or 'right'")
+    obs_range = pd.date_range(start=data_start, end=data_end, freq=freq,
+                              closed=closed)
+    return obs_range
+
+
+def _solpos_night_resample(observation, values):
+    # similar approach as in persistence_scalar_index.
+    # Calculate solar position and clearsky at 1 minute resolution to
+    # reduce errors from changing solar position during interval.
+    # Later, nighttime bools will be resampled over the interval
+    closed = datamodel.CLOSED_MAPPING[observation.interval_label]
+    freq = pd.Timedelta('1min')
+    interval_length = observation.interval_length
+    # need to calculate solar position for instants before or after the
+    # first/last labels depending on the interval label convention.
+    obs_range = _resample_date_range(interval_length, closed, freq, values)
+    # could add logic to remove points from obs_range where there are
+    # gaps in values. that would reduce computation time in some situations
+    solar_position = pvmodel.calculate_solar_position(
+        observation.site.latitude, observation.site.longitude,
+        observation.site.elevation, obs_range
+    )
+    # get the night flag as bitmask
+    night_flag = validator.check_irradiance_day_night_interval(
+        solar_position['zenith'],
+        closed,
+        interval_length,
+        solar_zenith_interval_length=freq,
+        _return_mask=True
+    )
+    # Better to use average solar position in downstream functions
+    # Best to return high res solar position and adapt downstream functions
+    # but that is left for future work
+    solar_position = solar_position.resample(
+        interval_length, closed=closed, label=closed
+    ).mean()
+    # return series with same index as input values
+    # i.e. put any gaps back in the data
+    night_flag = night_flag.loc[values.index]
```

Comment:
I feel like this warns about something and the suggested way to do this is to use ``.reindex()``. might be wrong though

Reply:
https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#indexing-with-list-with-missing-labels-is-deprecated

Reply:
If we get data with higher frequency than observation.interval_length, then `reindex` inserts NaNs and the series is coerced from `int` into `float`. This causes an error when `apply_*_validation` performs `|=` on the flags. So 9736521 sticks with `.loc` but reraises a more informative `KeyError`.

This situation came up in the broader test suite because the `report_objects` fixture happened to erroneously specify the observation interval length as 15 minutes instead of 1 minute. It shouldn't come up in production, but you could imagine that a user of the python library might run into it.